spark/src/main/scala/org/apache/spark/sql/delta/managedcommit/ManagedCommitUtils.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta.managedcommit
+
+import org.apache.spark.sql.delta.{DeltaLog, Snapshot}
+import org.apache.spark.sql.delta.util.FileNames
+import org.apache.spark.sql.delta.util.FileNames.{CompactedDeltaFile, DeltaFile, UnbackfilledDeltaFile}
+import org.apache.hadoop.fs.FileStatus
+
+object ManagedCommitUtils {
+
+  /**
+   * Returns an iterator of commit files starting from startVersion.
+   * If the iterator is consumed beyond what the file system listing shows, this method do a
+   * deltaLog.update() to find the latest version and returns listing results upto that version.
+   *
+   * @return an iterator of (file status, version) pair corresponding to commit files
+   */
+  def commitFilesIterator(
+      deltaLog: DeltaLog,
+      startVersion: Long): Iterator[(FileStatus, Long)] = {
+
+    def listDeltas(startVersion: Long, endVersion: Option[Long]): Iterator[(FileStatus, Long)] = {
+      deltaLog
+        .listFrom(startVersion)
+        .collect { case DeltaFile(fileStatus, version) => (fileStatus, version) }
+        .takeWhile { case (_, version) => endVersion.forall(version <= _) }
```

Comment:
Should we adopt `stopAt` here as well? I guess it's pretty unlikely to hit a listing page boundary right at the requested `endVersion`, so probably not worth the trouble?

Reply:
yes correct.